homeassistant/components/justnimbus/config_flow.py

```diff
@@ -0,0 +1,60 @@
+"""Config flow for JustNimbus integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import justnimbus
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_CLIENT_ID
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers import config_validation as cv
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_CLIENT_ID): cv.string,
+    },
+)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for JustNimbus."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user", data_schema=STEP_USER_DATA_SCHEMA
+            )
+
+        errors = {}
+
+        await self.async_set_unique_id(user_input[CONF_CLIENT_ID])
```

Comment:
Can the client id become invalid? In that case we'd need to migrate the unique id for the config entry, device entry and all entities, when doing re-authentication (not implemented yet). If that can happen it makes it a bit cumbersome to use the client id as unique id, I think.

An alternative is just to check for already created config entries with this client id, via `self._async_abort_entries_match`, and to use the config entry entry id as unique id for devices and entities.

Reply:
> Can the client id become invalid?

That one was in the review comments above, apparently, it can't.